app/templates/project-name.Gulp/tasks/styles.js

```diff
@@ -1,15 +1,16 @@
 import gulp from 'gulp';
 import sourcemaps from 'gulp-sourcemaps';
-import rename from 'gulp-rename';
 import plumber from 'gulp-plumber';
 import colors from 'ansi-colors';
 import log from 'fancy-log';
 import sass from 'gulp-sass';
 import postcss from 'gulp-postcss';
 import config from '../config';
 import autoprefixer from 'autoprefixer';
-import postcsstype from 'postcss-responsive-type';
-import postcssclear from 'postcss-clearfix';
+import postcssClear from 'postcss-clearfix';
+import postcssCssVars from 'postcss-css-variables';
+import postcssAssets from 'postcss-assets';
```

Comment:
Why do we want to use PostCSS Assets? While I see they make a point about isolating your source files from external changes, I hardly see it working.

Yes, images may change and chances are you need to go and check your styles anyway (what if an image has gotten too big).

Also it reminds me of Compass. They basically did exactly the same thing ([1](http://compass-style.org/reference/compass/helpers/inline-data/), [2](http://compass-style.org/reference/compass/helpers/image-dimensions/#image-width)).

When Compass grew unpopular, everyone got a number of projects where it was a real pleasure cleaning up those custom `@include` statements. Same here, but with custom CSS functions. PostCSS surely compiles it, but what if we switch to something else in the future.

I'm speaking only from my personal experience here, so if you guys think it's a good tool — let's use it.

Reply:
Lol, we need something new 🍅 Hehe.. On a serious note I partially agree with you regarding Compass and the pain there. I've used PostCSS Assets for numerous things. One converting old compass codebases. It doesn't bring some of the same things. But they weren't necessarily bad. Too bad it didn't work out for them. With that said I haven't used PostCSS Assets to its fullest and probably wouldn't ever but I am guilty of using `inline` nifty way to base64 encode and keep a reference to the original image. `Base path` and `cachebuster` has also been a nice addition.. at least untill we're still building via gulp. Well perhaps I just jumping guns.. let me know.. we can always give it a try and re-evaluate it later 😀

Reply:
Alright, you convinced me 😄 Let's give it a try 👍